lib/mayaUsd/fileio/transformWriter.cpp

```diff
@@ -260,6 +260,34 @@ bool UsdMayaTransformWriter::_GatherAnimChannel(
     return false;
 }
 
+void UsdMayaTransformWriter::_MakeAnimChannelsUnique(const UsdGeomXformable& usdXformable)
+{
```

Comment:
Can you move the `uniqueName()` function from `lib\mayaUsd\ufe\Utils.h` to `lib\mayaUsd\utils\util.h` and use that instead? You can call it as you go by keeping the `existingOps` set up to date after each rename.

This will make sure we have a unique way of making names unique in the app, and properly deal with names that already have a numeric suffix.

Reply:
Ah, I just checked uniqueName... unfortunately, the API is not quite compatible. uniqueName generates the "full name", but what we need to change is the piece used to generate the full name. (The suffix, to be precise.)

I will add the newly generated xform op to the existingOps set though as suggested, in case the newly added op collides with another one later on while iterating over all ops.

Reply:
Perhaps a simple-minded question: why do we append a separate translation?  Do we need to recover it separately?  Why not combine it with the existing translation?

Reply:
Well, the translation comes from different pieces of the rig, as far ass I could understand it. One came from a "foot" rig control, the other from a "foot-roll" rig-control. Given that they can control things is arbitrary ways, it seems hard to know how to merge them since there could be rotation, scale etc between two translations.

Reply:
Once the rig has been cached to USD, there is no longer a rig when edited-as-Maya, it is pure mesh/transform Maya nodes, so I expect they round-trip as part of our general support for generic xformop:order. If not, that code will need to be fixed, but we have no reason to think it should fail.

If we ever somehow support rig round-trip throw USD (via maybe a future more complex USD skeleton or somesuch) then we will have to revisit the issue, but that is not even on the horizon.